common/src/java/org/apache/hadoop/hive/common/auth/HiveAuthUtils.java

```diff
@@ -50,45 +49,108 @@
+  }
+
+  /**
+   * Create a TSocket for the provided host and port with specified loginTimeout and maxMessageSize.
+   * will default to Thrift library default.
+   * @param host Host to connect to.
+   * @param port Port to connect to.
+   * @param loginTimeout Socket timeout (0 means no timeout).
+   * @param maxMessageSize Size in bytes for max allowable Thrift message size, less than or equal to 0
+   *                       results in using the Thrift library default.
+   * @return TTransport TSocket for host/port
+   */
+  public static TTransport getSocketTransport(String host, int port, int loginTimeout, int maxMessageSize)
+      throws TTransportException {
+    TSocket tSocket = new TSocket(host, port, loginTimeout);
+    return configureThriftMaxMessageSize(tSocket, maxMessageSize);
+  }
+
+  public static TTransport getSSLSocket(String host, int port, int loginTimeout, TSSLTransportParameters params,
+      int maxMessageSize) throws TTransportException {
+    // The underlying SSLSocket object is bound to host:port with the given SO_TIMEOUT and
+    // SSLContext created with the given params
+    TSocket tSSLSocket = null;
+    if (params != null) {
+      tSSLSocket = TSSLTransportFactory.getClientSocket(host, port, loginTimeout, params);
+    } else {
+      tSSLSocket = TSSLTransportFactory.getClientSocket(host, port, loginTimeout);
+    }
+    configureThriftMaxMessageSize(tSSLSocket, maxMessageSize);
+    return getSSLSocketWithHttps(tSSLSocket, maxMessageSize);
+  }
+
+  public static TTransport getSSLSocket(String host, int port, int loginTimeout, String trustStorePath,
+      String trustStorePassWord, String trustStoreType, String trustStoreAlgorithm) throws TTransportException {
+    return getSSLSocket(host, port, loginTimeout, trustStorePath, trustStorePassWord, trustStoreType,
+        trustStoreAlgorithm, -1);
   }
 
-  public static TTransport getSSLSocket(String host, int port, int loginTimeout)
-    throws TTransportException {
-    // The underlying SSLSocket object is bound to host:port with the given SO_TIMEOUT
-    TSocket tSSLSocket = TSSLTransportFactory.getClientSocket(host, port, loginTimeout);
-    return getSSLSocketWithHttps(tSSLSocket);
+  public static TTransport getSSLSocket(String host, int port, int loginTimeout) throws TTransportException {
+    return getSSLSocket(host, port, loginTimeout, -1);
   }
 
-  public static TTransport getSSLSocket(String host, int port, int loginTimeout,
-      String trustStorePath, String trustStorePassWord, String trustStoreType,
-      String trustStoreAlgorithm) throws TTransportException {
-    TSSLTransportFactory.TSSLTransportParameters params =
-      new TSSLTransportFactory.TSSLTransportParameters();
+  public static TTransport getSSLSocket(String host, int port, int loginTimeout, int maxMessageSize)
+      throws TTransportException {
+    return getSSLSocket(host, port, loginTimeout, null, maxMessageSize);
```

Comment:
For the default values being passed (e.g null) or -1 etc.  could you add a /* param name */